tests/e2e/utilities/setup-and-go.ts

```diff
@@ -20,46 +20,74 @@ export async function setupAndGo(
     allowGeo?: boolean;
     amUrl?: string;
     clientId?: string;
+    dialogInput?: string;
     email?: string;
     pw?: string;
     realmPath?: string;
     resourceUrl?: string;
+    selector?: string;
     scope?: string;
     tokenStore?: string;
     tree?: string;
     un?: string;
   },
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-): Promise<{ browser: any; page: any }> {
+): Promise<{ browser: any; messageArray: string[]; networkArray: string[] }> {
   const browser = await browsers[browserType].launch({ headless: true });
-  const context = await browser.newContext({
-    bypassCSP: true,
-    geolocation: { latitude: 24.9884, longitude: -87.3459 },
-    ignoreHTTPSErrors: true,
-    permissions: config && config.allowGeo ? ['geolocation'] : [],
-  });
-  const page = await context.newPage();
+  const messageArray = [];
+  const networkArray = [];
 
-  const url = new URL(`${BASE_URL}/${path}`);
+  // If anything fails, ensure we close the browser to end the process
+  try {
```

Comment:
This try-catch wraps everything to catch for thrown errors and closes the browser to ensure proper clean up in the event of a thrown error.